File: split/freenode.pas
procedure freenode(p: halfword; s: halfword);
var
  q: halfword;
begin
  mem[p].hh.lh := s;
  mem[p].hh.rh := 65535;
  q := mem[rover + 1].hh.lh;
  mem[p + 1].hh.lh := q;
  mem[p + 1].hh.rh := rover;
  mem[rover + 1].hh.lh := p;
  mem[q + 1].hh.rh := p;{varused:=varused-s;}

end;

File: cpp/freenode.cpp
#include "freenode.h"

void freenode(halfword p, halfword s)
{
	info(p) = s;
	link(p) = 0xFF'FF;
	halfword q = info(rover+1);
	info(p+1) = q;
	link(p+1) = rover;
	info(rover+1) = p;
	link(q+1) = p;
}
